### ISSUE TITLE
Bug: Table prefixes doesn't work with insert method.

### ISSUE THREAD

**Issue by userlond** [robmorgan/phinx#714]
Hi, I tave db config with table prefixes:
```php
<?php
// this file contains DB_* constants
require 'www/config.php';

return [
    'paths' => [
        'migrations' => 'db/migrations',
        'seeds' => 'db/seeds',
    ],
    'environments' => [
        'default_migration_table' => DB_PREFIX . 'phinxlog',
        'default_database' => 'db',
        'db' => [
            "adapter" => 'mysql',
            'table_prefix' => DB_PREFIX,
            "host" => DB_HOSTNAME,
            "name" => DB_DATABASE,
            "user" => DB_USERNAME,
            "pass" => DB_PASSWORD,
            "port" => DB_PORT,
        ],
    ],
];
```

My migration is below, issues are marked via comments:

```php
<?php

use Phinx\Migration\AbstractMigration;

class AddWatermarkMigration extends AbstractMigration
{
    public function up()
    {
        $tableAdapter = new \Phinx\Db\Adapter\TablePrefixAdapter($this->getAdapter());

        // Here prefix works
        if (!$this->hasTable('watermark_settings')) {
            // Here prefix also works 
            $this->table('watermark_settings')
                ->addColumn('key', 'string', ['limit' => 255])
                ->addColumn('value', 'string', ['limit' => 1024])
                ->create();

            // It seems, that table prefix does't work with insert statement.
            // $this->table('watermark_settings')->insert([...]) raises PDO table doesn't exists error.
            // So I've added table prefix manually via $tableAdapter.
            // Now it works, but I think It's tricky
            $this->table($tableAdapter->getAdapterTableName('watermark_settings'))
                ->insert([
                    [
                        'key' => 'active',
                        'value' => '1',
                    ],
                ])
                ->save();
        }
    }
}
```

So I thing `$this->table('table_name')->insert([..])` should take table prefixes in account, don't you?

**Comment by robmorgan** [comment]
@userlond yes it should, currently it doesn't. I'll make a fix in an upcoming release.

**Comment by robmorgan** [comment]
fixed in 129f2ad56abf24084096868a8bb3e72d0c464b9b.

**Comment by userlond** [comment]
:+1:
